### ISSUE TITLE
Developer tools: does not pause execution at `debugger`

### ISSUE THREAD

**Issue by rerdos** [electron/electron#4842]
* Electron version: 0.37.2
* Operating system: OS X (El Capitan 10.11.3)

If you use `debugger` it will not pause the execution of the program. If I downgrade to electron version `0.36.x`, then it works.

Same happens when you set up breakpoints in the Developer Tools.

![Debugger issue]
(http://i.imgur.com/jps3VVH.gif)

**Comment by christoth** [comment]
I'm experiencing the same issue with execution not stopping on breakpoints in Developer Tools on 0.37.2 on Win7 x64

**Comment by MikeGCV** [comment]
This problem is happening to me also on Win10 x64 electron 0.37.2, on 0.36.11 debug works fine after refresh

**Comment by paul321** [comment]
Same problem if i refresh the app using CTRL + R or CMD+R debugger stops working, also the CPU load it's very high when the debugger stops for the first time

**Comment by steelbrain** [comment]
I am also experiencing this, even break points do not work

**Comment by darrrk** [comment]
Confirm this on 0.37.3

**Comment by pravdomil** [comment]
Not working for me, v 0.37.3 on OS X 10.11.3. Trying set breakpoint manually and via debugger;

**Comment by tomzx** [comment]
It appears the issue is more generalized, namely no breakpoint is triggered. I've been able to reproduce this issue on both Win7 x64 and Mac OS X 10.10.2, for v0.37.2 and v0.37.3. v0.36.12 works fine.

**Comment by bbondy** [comment]
Confirming what @tomzx wrote.

**Comment by MikeGCV** [comment]
Ok, i have founde something today that may help, if i install 0.36.12, use break points, and then install 0.37.4 without deleting the userdata folder, the old break points from 0.36.12 work with reload, but the new ones marked on 0.37.4 don't. They only work on restart off the application. Hope this info helps

**Comment by remss** [comment]
Same issue (electron 0.37.4, window7 and archlinux), breakpoints are broken. Stuck with 0.36 :-(

**Comment by pravdomil** [comment]
electron v0.37.5 mac - issue persists

**Comment by darrrk** [comment]
Why less activity here? It's high priority issue, that block most developers to move from 0.36.x

**Comment by Booster2ooo** [comment]
Same in 0.37.5 (win7 x64)

**Comment by MikeGCV** [comment]
Thank you

**Comment by bpasero** [comment]
@zcbenz would it be possible to get a new release with this fix? we recently switched to Electron 0.37.5 for VS Code *next* and our development experience is a bit downgraded because changing a file on disk and reloading from dev tools disables all breakpoints in those files.

**Comment by jefffriesen** [comment]
This ticket is closed but I'm not sure it should be. I did the simplest possible `debugger` test in electron-quick-start and the app still did not pause on the debugger:
https://github.com/electron/electron-quick-start/compare/master...jefffriesen:debugger

1. Updated to Electron 1.4.1
2. Put a debugger in renderer.js:
```js
console.log('before debugger')
debugger
console.log('after debugger')
```
Execution is never paused and both console.log's fire.

So either this isn't fixed in 1.4.1 or maybe devtools needs to be configured differently? Or something else?

I'm trying to fix this https://github.com/chentsulin/electron-react-boilerplate/issues/379

Any help would be awesome. Thanks

**Comment by greggman** [comment]
I'm hitting this as well in 1.4.4. My workaround so far is been to add a setTimeout before I do anything **AND** that timeout has to be > 500ms.  Not sure what the shortest value is but at 500ms the debugger doesn't catch. If I wait 2000ms then the dubugger does catch.

In other words 

```
setTimeout(() => {
  console.log('before debugger');
  debugger;
  console.log('after debugger');
}, 2000);
```

Is there something about the debugger needing to settle down before it can catch breakpoints?
